jetty-core/jetty-websocket/jetty-websocket-core-common/src/main/java/org/eclipse/jetty/websocket/core/internal/FrameFlusher.java

```diff
@@ -511,27 +516,61 @@ public String toString()
         return String.format("%s[queueSize=%d,aggregate=%s]",
             super.toString(),
             getQueueSize(),
-            batchBuffer);
+            _batchBuffer);
     }
 
-    private static class Entry extends FrameEntry
+    private class Entry extends FrameEntry implements CyclicTimeouts.Expirable
     {
-        private final long timeOfCreation = System.currentTimeMillis();
+        private long _expiry = Long.MAX_VALUE;
 
         private Entry(Frame frame, Callback callback, boolean batch)
         {
             super(frame, callback, batch);
+
+            long currentTime = NanoTime.now();
+            if (_frameTimeout > 0)
+            {
+                long frameExpiry = Math.addExact(currentTime, TimeUnit.MILLISECONDS.toNanos(_frameTimeout));
+                _expiry = nanoTimeMin(_expiry, frameExpiry);
```

Comment:
`_expiry` is always `Long.MAX_VALUE` before the call to `nanoTimeMin()` so I fail to see why you're making a comparison, it looks like `_expiry = frameExpiry;` is what you should be doing.

Plus calling `NanoTime.isBeforeOrSame()` with the 1st parameter not being a nanotime is going to return a result at random.